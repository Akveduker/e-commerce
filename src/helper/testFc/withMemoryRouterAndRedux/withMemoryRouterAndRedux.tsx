import { configureStore } from "@reduxjs/toolkit"
import { render } from "@testing-library/react"
import { InitialEntry } from "history"
import { Provider } from "react-redux"
import { MemoryRouter } from "react-router-dom"
import { mockReducersForStore } from "../../mock/mockStore/mockStore"
export const withMemoryRouterAndRedux = (children: React.ReactNode, reducers?: Partial<typeof mockReducersForStore>, initialEntries?: InitialEntry[]) => {
    const mockStore = configureStore({
        reducer: { ...mockReducersForStore, ...reducers }
    });
    return render(
        <MemoryRouter initialEntries={initialEntries}>
            <Provider store={mockStore}>

                {children}
            </Provider>
        </MemoryRouter>
    )
}