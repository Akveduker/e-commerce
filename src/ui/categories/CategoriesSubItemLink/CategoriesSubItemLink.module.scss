@import "../../../styles/_vars.scss";
@import "../../../styles/_mixins.scss";
.link {
    color: $C1A;
    @include fontMedium;
    transition: $linkTranstion;
    &:hover {
        color: $C2A;
    }
}
