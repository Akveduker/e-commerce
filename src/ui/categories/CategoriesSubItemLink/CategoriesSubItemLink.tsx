import React, { FC } from 'react';
import { Link, LinkProps } from 'react-router-dom';
import { roots } from '../../../router/routes/routes';
import { useAppSelector } from '../../../store/store';
import { findSingleCategoryById } from '../../../utils/store/categories/findSingleCategoryById/findSingleCategoryById';
import { getCategoriesfromStore } from '../../../utils/store/categories/getCategoriesfromStore';
import s from './CategoriesSubItemLink.module.scss'
interface CategoriesSubItemLinkProps {
    id: number
}
const CategoriesSubItemLink: FC<CategoriesSubItemLinkProps> = ({ id }) => {
    const category = findSingleCategoryById(id, useAppSelector(getCategoriesfromStore).categories)
    if (!category) return null
    return (
        <Link className={s['link']} to={`${roots.main}${roots.categories}/${id}`} data-testid={'link'}>
            {category.name}
        </Link>
    );
};

export default CategoriesSubItemLink;