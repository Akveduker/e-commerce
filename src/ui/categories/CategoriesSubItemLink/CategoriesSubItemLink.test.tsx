import { render, screen } from "@testing-library/react"
import { withMemoryRouterAndRedux } from "../../../helper/testFc/withMemoryRouterAndRedux/withMemoryRouterAndRedux"
import CategoriesSubItemLink from "./CategoriesSubItemLink"

describe('CategoriesSubItemLink', () => {
    test('Категория есть в сторе', () => {
        const { container } = withMemoryRouterAndRedux(<CategoriesSubItemLink id={1} />)
        const link = screen.getByTestId('link')
        expect(container).toMatchSnapshot()
        expect(link).toHaveAttribute('href', '/categories/1')
        expect(link).toBeHre
    })
    test('Категории нет в сторе', () => {
        const { container } = withMemoryRouterAndRedux(<CategoriesSubItemLink id={100} />)
        expect(container).toMatchSnapshot()
    })
})