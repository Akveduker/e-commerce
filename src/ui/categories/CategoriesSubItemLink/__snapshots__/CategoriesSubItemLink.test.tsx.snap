// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesSubItemLink Категории нет в сторе 1`] = `<div />`;

exports[`CategoriesSubItemLink Категория есть в сторе 1`] = `
<div>
  <a
    class="link"
    data-testid="link"
    href="/categories/1"
  >
    name
  </a>
</div>
`;
